core/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -435,6 +436,21 @@ public void deleteSnapshot(SnapshotId snapshotId) {
                     }
                 }
             }
+
+            // cleanup indices that are no longer part of the repository
+            final Collection<IndexId> indicesToCleanUp = Sets.newHashSet(repositoryData.getIndices().values());
+            indicesToCleanUp.removeAll(updatedRepositoryData.getIndices().values());
+            final BlobContainer indicesBlobContainer = blobStore().blobContainer(basePath().add("indices"));
+            for (final IndexId indexId : indicesToCleanUp) {
+                try {
+                    indicesBlobContainer.deleteBlob(indexId.getId());
+                } catch (IOException e) {
+                    // if the directory isn't empty for some reason, it will fail to clean up;
```

Comment:
Can we catch DirectoryNotEmptyException separatly in addition to IOException and log both exception as at least DEBUG level?